runtime/queries/proto.go

```diff
@@ -62,16 +62,18 @@ func ProtoToQuery(q *runtimev1.Query, attrs map[string]any) (runtime.Query, erro
 }
 
 // ProtoFromJSON builds a proto query from a query name, JSON args, and optional execution time.
-func ProtoFromJSON(qryName, qryArgsJSON string, executionTime *time.Time) (*runtimev1.Query, error) {
+func ProtoFromJSON(qryName, qryArgsJSON string, executionTime *time.Time) (*runtimev1.Query, string, error) {
```

Comment:
Can we split this out into a `func MetricsViewFromQuery(qry *runtimev1.Query) (name string, ok bool)` function instead? It would need to reproduce the `switch`, but would nicely separate the concerns (e.g. in case we need to use `ProtoFromJSON` for non-metrics queries in the future.)